### PR TITLE
feat: add CPE format validation in property setter

### LINKED ISSUE #580: Improvement: Apply Regex check to `Component.cpe`
The CycloneDX scpecification defines a Regex for `Component.cpe`, but this library does not enforce this.

### PR DESCRIPTION
**Pull request by saquibsaifee**
Fixes #580 

- Implemented regex-based validation for CPE format in the model.
- Added tests to verify the handling of invalid CPE strings.

### CODE REVIEW

**Comment by saquibsaifee** [comment]
@madpah and @jkowalleck please advice if these changes are enough to add this validation.

File: cyclonedx/model/component.py [hunk 1]
```diff
@@ -1457,6 +1467,8 @@ def cpe(self) -> Optional[str]:
 
     @cpe.setter
     def cpe(self, cpe: Optional[str]) -> None:
+        if cpe and not CPE_REGEX.fullmatch(cpe):
+            raise ValueError(f'Invalid CPE format: {cpe}')
```
Comment (jkowalleck) [inline cyclonedx/model/component.py:1471]: this behavioral change is a breaking change. 

this fact is not a blocker.

**Review by jkowalleck — CHANGES_REQUESTED** [review]
CPE is a complex, external spec - outside the domain of CycloneDX.
This fact leads me to the architectural decision: we do not want to maintain an implementation of this external spec in the domain of CycloneDX python library.

meaning: this PR will not be accepted in its current form.

we might consider a usage of an external library, like <https://pypi.org/project/cpe/>.

**Comment by jkowalleck** [comment]
Thank you for the PR, @saquibsaifee .

I will close this PR for reasons detailed in https://github.com/CycloneDX/cyclonedx-python-lib/pull/706#pullrequestreview-2366402891

For a different approach, please open a new pullrequest.

**Pull request closed without merging**